src/int/from.rs

```diff
@@ -0,0 +1,156 @@
+//! `From`-like conversions for [`Int`].
+
+use crate::{Int, Limb, Uint, Word, I128, I64};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Create a [`Int`] from an `i8` (const-friendly)
+    // TODO(tarcieri): replace with `const impl From<i8>` when stable
+    pub const fn from_i8(n: i8) -> Self {
+        assert!(LIMBS >= 1, "number of limbs must be greater than zero");
+        Int::new_from_uint(Uint::new([Limb(n as Word)])).resize()
+    }
+
+    /// Create a [`Int`] from an `i16` (const-friendly)
+    // TODO(tarcieri): replace with `const impl From<i16>` when stable
+    pub const fn from_i16(n: i16) -> Self {
+        assert!(LIMBS >= 1, "number of limbs must be greater than zero");
+        Int::new_from_uint(Uint::new([Limb(n as Word)])).resize()
+    }
+
+    /// Create a [`Int`] from an `i32` (const-friendly)
+    // TODO(tarcieri): replace with `const impl From<i32>` when stable
+    pub fn from_i32(n: i32) -> Self {
+        assert!(LIMBS >= 1, "number of limbs must be greater than zero");
+        Int::new_from_uint(Uint::new([Limb(n as Word)])).resize()
+    }
+
+    /// Create a [`Int`] from an `i64` (const-friendly)
+    // TODO(tarcieri): replace with `const impl From<i64>` when stable
+    #[cfg(target_pointer_width = "32")]
+    pub const fn from_i64(n: i64) -> Self {
+        Int::new_from_uint(Uint::<{ I64::LIMBS }>::from_u64(n as u64)).resize()
```

Comment:
this should work and is a little simpler:
Int::new_from_uint(Uint::from_u64(n as u 64))

Reply:
I don't think that would work. Let me explain why:

In scaling up a negative integer, we have to make sure that all new limbs are initialized with `Limb::MAX` instead of `Limb::ZERO` (as is the default for `Uint`). So, what I'm doing here is:
- cast `i64` to `u64`,
- ingest this `u64` into a `Uint` that is as small as possible,
- wrap the `Uint` into an `Int`, and
- scale up the `int` (which I wrote to make sure it uses the right `Limb` type).

IIUC, `Int::new_from_uint(Uint::from_u64(n as u 64))` does
- cast `i64` to `u64`,
- ingest this `u64` into a `Uint<LIMBS>`, filling any missing limbs with `Limb::ZERO`, and
- wrapping the `Uint` into an `Int`.

This would not work for negative values.

Would you agree?

Reply:
ahh, right,
I forgot that you can call from_i64 to Int with more than two limbs :)